E01-Geometry/Dovydas/src/Kilo/Square.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Kilo;
+class Square implements Shape
+{
+    private float $length;
+    private float $height;
+    private Geometry $geometry;
+
+    /**
+     * Square constructor.
+     * @param float $length
+     * @param float $height
+     */
+    public function __construct(float $length, float $height)
+    {
+        $this->length = $length;
+        $this->height = $height;
+        $this->setGeometry();
```

Comment:
This Circular dependancy is a bit over-engineering and it's a circular dependancy. 
Your code would fail using Service dependancy.

I would suggest moving area and perimeter calculation into responsibility of `Shape` model.